UKHomeOffice/form-api-server
# COP-9765: Fix critical issue dependencies

### Description
Not using `lint-staged` at all so removed it. And upgraded the version of `nyc` to resolve the critical issue with one of its dependencies. The new version runs fine locally and produces a coverage report identical to the previous version.

### To test
The last Drone build failed due to critical issues. If it succeeds once this is merged, it can be considered a success - both changes are dev dependencies so there should be no functional change.
